ldk/javascript/src/search/index.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  promisifyMappedWithTwoParams,
+  promisifyMappedWithThreeParams,
+  promisifyWithParam,
+} from '../promisify';
+import * as mapper from './mapper';
+import { Document, Config, SearchResult } from './types';
+
+export * from './types';
+
+export interface Index {
+  /**
+   * Executes a search with the search term over the data.
+   * @param term - The term that will be used to search over the index.
+   * @returns A Promise resolving with the search results as a string
+   */
+  search(term: string): Promise<SearchResult>;
+  /**
+   * Executes a search with the using a query string to search over the data.
+   *
+   * @param queryString - The query string that will be used to search over the index.
+   * @returns A Promise resolving with the search results as a string
+   */
+  queryStringSearch(queryString: string): Promise<SearchResult>;
+  /**
+   * Executes a search with the using a query string to search over the data.
+   *
+   * @param documents - The date that will be indexed and searchable
+   * @param config - The configuration object defines how the index will preform certain searches
+   */
+  update(documents: Document[], config: Config): Promise<void>;
```

Comment:
It is a little confusing to me how `Document[]` maps to some kind of date. Do you mean "data"?

Reply:
Should of been `data`